### PR TITLE
SONARJAVA-4403 Add exception to S3553

### PR DESCRIPTION
**Pull request by johann-beleites-sonarsource**
Don't flag Optional uses as parameter types when the method is overriding

### CODE REVIEW

File: java-checks/src/main/java/org/sonar/java/checks/OptionalAsParameterCheck.java
Comment (alban-auzeill) [inline java-checks/src/main/java/org/sonar/java/checks/OptionalAsParameterCheck.java:54]: Interview question: What is the result of this java expression?
```java
(new Boolean(false) == Boolean.FALSE)
```
So here, because we are not `kotlin` developers :stuck_out_tongue: but `java` developers :grin:, we should use:
```java
Boolean.FALSE.equals(methodTree.isOverriding())
```

Except for this, I approve this PR, once fixed you can merge.

**Review by alban-auzeill — APPROVED** [review]

File: java-checks/src/main/java/org/sonar/java/checks/OptionalAsParameterCheck.java
Comment (johann-beleites-sonarsource) [reply java-checks/src/main/java/org/sonar/java/checks/OptionalAsParameterCheck.java:54]: :eyes:

**Pull request merged**
